File: sdk/php/src/Resources/Proxmox.php
<?php

namespace HiTechCloud\SDK\Resources;

class Proxmox extends BaseResource
{
    /**
     * Get status
     */
    public function getStatus(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpve/status');
    }

    /**
     * Power action
     */
    public function powerAction(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/htcpve/power', $data);
    }

    /**
     * List VMs
     */
    public function listVms(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpve/vms');
    }

    /**
     * List IPs
     */
    public function listIps(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpve/ips');
    }

    /**
     * Set rDNS
     */
    public function setRdns(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/htcpve/rdns', $data);
    }

    /**
     * List backups
     */
    public function listBackups(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpve/backups');
    }

    /**
     * Create backup
     */
    public function createBackup(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/htcpve/backups', $data);
    }

    /**
     * List snapshots
     */
    public function listSnapshots(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpve/snapshots');
    }

    /**
     * Create snapshot
     */
    public function createSnapshot(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/htcpve/snapshots', $data);
    }

    /**
     * Get bandwidth usage
     */
    public function getBandwidthUsage(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpve/usage');
    }

}

File: sdk/csharp/Resources/Proxmox.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class Proxmox : BaseResource
{
    public Proxmox(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> StatusAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpve/status", id));
    }

    public async Task<Dictionary<string, object?>> PowerAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/htcpve/power", id), data);
    }

    public async Task<Dictionary<string, object?>> VmsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpve/vms", id));
    }

    public async Task<Dictionary<string, object?>> IpsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpve/ips", id));
    }

    public async Task<Dictionary<string, object?>> SetRdnsAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/htcpve/rdns", id), data);
    }

    public async Task<Dictionary<string, object?>> BackupsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpve/backups", id));
    }

    public async Task<Dictionary<string, object?>> CreateBackupAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/htcpve/backups", id), data);
    }

    public async Task<Dictionary<string, object?>> SnapshotsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpve/snapshots", id));
    }

    public async Task<Dictionary<string, object?>> CreateSnapshotAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/htcpve/snapshots", id), data);
    }

    public async Task<Dictionary<string, object?>> UsageAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpve/usage", id));
    }
}
